service/storage_proxy.cc

```diff
@@ -398,7 +437,9 @@ class storage_proxy::remote {
             netw::messaging_service::msg_addr src_addr, rpc::opt_time_point t,
             auto schema_version, auto in, inet_address_vector_replica_set forward, gms::inet_address reply_to,
             unsigned shard, storage_proxy::response_id_type response_id, std::optional<tracing::trace_info> trace_info,
-            auto&& apply_fn1, auto&& forward_fn1) {
+            std::optional<fencing_token> fence,
```

Comment:
We could declare that token 0 is "no token" and so get rid of the optionals, streamlining the code somewhat.